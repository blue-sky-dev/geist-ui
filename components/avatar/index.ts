import Avatar from './avatar'
import { AvatarProps } from './avatar'

export type Props = AvatarProps
export default Avatar
