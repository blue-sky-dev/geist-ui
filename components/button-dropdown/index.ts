import ButtonDropdown from './button-dropdown'
import ButtonDropdownItem from './item'

ButtonDropdown.Item = ButtonDropdownItem

export default ButtonDropdown
