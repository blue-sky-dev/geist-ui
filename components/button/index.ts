import Button from './button'
import { ButtonProps } from './button'

export type Props = ButtonProps
export default Button
