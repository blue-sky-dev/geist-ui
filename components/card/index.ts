import Card from './card'
import { CardProps } from './card'

export type Props = CardProps
export default Card
