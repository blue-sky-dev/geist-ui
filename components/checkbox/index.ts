import Checkbox from './checkbox'
import CheckboxGroup from './checkbox-group'
import { CheckboxProps, CheckboxEvent } from './checkbox'

export type Props = CheckboxProps
export type Event = CheckboxEvent

Checkbox.Group = CheckboxGroup

export default Checkbox
