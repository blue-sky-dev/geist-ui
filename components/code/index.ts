import Code from './code'
import { CodeProps } from './code'

export type Props = CodeProps
export default Code
