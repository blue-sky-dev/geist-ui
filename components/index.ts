/// <reference types="styled-jsx" />

export { default as useTheme } from './styles/use-theme'
export { default as ZEITUIProvider } from './providers/zeit-ui-provider'
export { default as useToasts } from './toast'
export { default as useInput } from './input/use-input'
export { default as useModal } from './modal/use-modal'
export { default as useTabs } from './tabs/use-tabs'
export { default as CSSBaseline } from './styles/css-baseline'
export { default as Avatar } from './avatar'
export { default as Text } from './text'
export { default as Note } from './note'
export { default as Link } from './link'
export { default as Col } from './col'
export { default as Row } from './row'
export { default as Container } from './row'
export { default as Button } from './button'
export { default as Card } from './card'
export { default as Code } from './code'
export { default as Display } from './display'
export { default as Description } from './description'
export { default as Image } from './image'
export { default as Spacer } from './spacer'
export { default as Tag } from './tag'
export { default as Dot } from './dot'
export { default as Keyboard } from './keyboard'
export { default as Checkbox } from './checkbox'
export { default as Fieldset } from './fieldset'
export { default as Modal } from './modal'
export { default as Spinner } from './spinner'
export { default as ButtonDropdown } from './button-dropdown'
export { default as Capacity } from './capacity'
export { default as Input } from './input'
export { default as Radio } from './radio'
export { default as Select } from './select'
export { default as Tabs } from './tabs'
export { default as Progress } from './progress'
export { default as Tree } from './file-tree'
export { default as Badge } from './badge'
export { default as AutoComplete } from './auto-complete'
export { default as Collapse } from './collapse'
