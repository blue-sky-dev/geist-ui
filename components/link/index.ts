import Link from './link'
import { LinkProps } from './link'

export type Props = LinkProps
export default Link
