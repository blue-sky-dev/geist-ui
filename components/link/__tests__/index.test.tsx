import React from 'react'
import { mount } from 'enzyme'
import { Link } from 'components'

describe('Link', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <div>
        <Link href="https://react.zeit-ui.co">link</Link>
        <Link href="https://react.zeit-ui.co" color>
          link
        </Link>
        <Link href="https://react.zeit-ui.co" icon>
          link
        </Link>
        <Link href="https://react.zeit-ui.co" underline>
          link
        </Link>
        <Link href="https://react.zeit-ui.co" block>
          link
        </Link>
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be no errors when href missing', () => {
    const errorSpy = jest.spyOn(console, 'error')
    const wrapper = mount(<Link />)
    expect(errorSpy).not.toHaveBeenCalled()
    expect(() => wrapper.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })

  it('should forward ref', () => {
    let ref = React.createRef<HTMLAnchorElement>()
    const errorSpy = jest.spyOn(console, 'error')
    const wrapper = mount(<Link ref={ref} />)

    expect(errorSpy).not.toHaveBeenCalled()
    expect(ref.current).not.toBeNull()
    expect(() => wrapper.unmount()).not.toThrow()
    errorSpy.mockRestore()
  })

  it('an warning should be thrown when using the pure prop', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
    mount(<Link pure />)
    expect(errorSpy).toHaveBeenCalled()
    errorSpy.mockRestore()
  })
})
