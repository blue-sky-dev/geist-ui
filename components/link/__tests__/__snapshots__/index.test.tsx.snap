// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
"<div><a class=\\"link \\" href=\\"https://react.zeit-ui.co\\">link<svg viewBox=\\"0 0 24 24\\" width=\\"1em\\" height=\\"1em\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" class=\\"icon\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path><style>
        .icon {
          margin: 0 5px;
          display: inline-flex;
          align-self: center;
        }
      </style></svg><style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: inherit;
            text-decoration: none;
            padding: calc(0 * 0.8) calc(0 * 1.7);
            border-radius: 0;
            width: fit-content;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            background-color: unset;
          }
        </style></a><a class=\\"link \\" href=\\"https://react.zeit-ui.co\\">link<svg viewBox=\\"0 0 24 24\\" width=\\"1em\\" height=\\"1em\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: rgb(0, 112, 243);\\" class=\\"icon\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path><style>
        .icon {
          margin: 0 5px;
          display: inline-flex;
          align-self: center;
        }
      </style></svg><style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: #0070f3;
            text-decoration: none;
            padding: calc(0 * 0.8) calc(0 * 1.7);
            border-radius: 0;
            width: fit-content;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            background-color: unset;
          }
        </style></a><a class=\\"link \\" href=\\"https://react.zeit-ui.co\\">link<style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: inherit;
            text-decoration: none;
            padding: calc(0 * 0.8) calc(0 * 1.7);
            border-radius: 0;
            width: fit-content;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            background-color: unset;
          }
        </style></a><a class=\\"link \\" href=\\"https://react.zeit-ui.co\\">link<svg viewBox=\\"0 0 24 24\\" width=\\"1em\\" height=\\"1em\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" class=\\"icon\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path><style>
        .icon {
          margin: 0 5px;
          display: inline-flex;
          align-self: center;
        }
      </style></svg><style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: inherit;
            text-decoration: none;
            padding: calc(0 * 0.8) calc(0 * 1.7);
            border-radius: 0;
            width: fit-content;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: underline;
          }

          .link:hover {
            background-color: unset;
          }
        </style></a><a class=\\"link \\" href=\\"https://react.zeit-ui.co\\">link<svg viewBox=\\"0 0 24 24\\" width=\\"1em\\" height=\\"1em\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: rgb(0, 112, 243);\\" class=\\"icon\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path><style>
        .icon {
          margin: 0 5px;
          display: inline-flex;
          align-self: center;
        }
      </style></svg><style>
          .link {
            display: inline-flex;
            align-items: baseline;
            line-height: inherit;
            color: #0070f3;
            text-decoration: none;
            padding: calc(4pt * 0.8) calc(4pt * 1.7);
            border-radius: 5px;
            width: fit-content;
          }

          .link:hover,
          .link:active,
          .link:focus {
            text-decoration: none;
          }

          .link:hover {
            background-color: #0076ff1a;
          }
        </style></a></div>"
`;
